File: src/main/java/kr/co/lunatalk/domain/payment/dto/response/PaymentCancelResponse.java
package kr.co.lunatalk.domain.payment.dto.response;

import io.swagger.v3.oas.annotations.media.Schema;
import kr.co.lunatalk.domain.order.domain.Order;
import kr.co.lunatalk.domain.order.domain.OrderStatus;
import kr.co.lunatalk.domain.payment.domain.Payment;
import kr.co.lunatalk.domain.payment.domain.PaymentStatus;

public record PaymentCancelResponse(
	@Schema(description = "주문 번호", example = "L3ABCDEFG")
	String orderNumber,
	@Schema(description = "결제 상태", example = "CANCELLED")
	PaymentStatus paymentStatus,
	@Schema(description = "주문 상태", example = "CANCELLED")
	OrderStatus orderStatus
) {

	public static PaymentCancelResponse of(Order order, Payment payment) {
		return new PaymentCancelResponse(
			order.getOrderNumber(),
			payment.getStatus(),
			order.getStatus()
		);
	}
}

File: lunatalk-domain/src/main/kotlin/kr/co/lunatalk/domain/payment/dto/response/PaymentCancelResponse.kt
package kr.co.lunatalk.domain.payment.dto.response

import io.swagger.v3.oas.annotations.media.Schema
import kr.co.lunatalk.domain.order.domain.Order
import kr.co.lunatalk.domain.order.domain.OrderStatus
import kr.co.lunatalk.domain.payment.domain.Payment
import kr.co.lunatalk.domain.payment.domain.PaymentStatus

data class PaymentCancelResponse(
    @Schema(description = "주문 번호", example = "L3ABCDEFG")
    val orderNumber: String?,

    @Schema(description = "결제 상태", example = "CANCELLED")
    val paymentStatus: PaymentStatus?,

    @Schema(description = "주문 상태", example = "CANCELLED")
    val orderStatus: OrderStatus?
) {
    companion object {
        fun of(order: Order, payment: Payment): PaymentCancelResponse {
            return PaymentCancelResponse(
                orderNumber = order.orderNumber,
                paymentStatus = payment.status,
                orderStatus = order.status
            )
        }
    }
}
